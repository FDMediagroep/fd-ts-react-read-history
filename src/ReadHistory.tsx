import React, { PureComponent } from "react";
import Card, { CardStyle } from "@fdmg/fd-card";
import TypoGraphy, { getAllTextStyles } from "@fdmg/fd-typography";
import { createGlobalStyle } from "styled-components";

interface NewsItem {
    uuid: string;
    dateTime: string;
    isRead?: boolean;
    target?: string;
    title: string;
    url: string;
}

export interface Props {
    className?: string;
    news: NewsItem[];
    title?: string;
}

export default class ReadHistory extends PureComponent<Props, any> {

    render() {
        return (
            <>
                <GlobalStyle/>
                <Card className={`fd-card-read-history${this.props.className ? ` ${this.props.className}` : ''}`}>
                    <TypoGraphy className="h-read-history" textStyle='card-h'><h3>{this.props.title ? this.props.title : 'Recent gelezen'}</h3></TypoGraphy>
                    <nav>
                        {
                            this.props.news.map((newsItem) => {
                                return (
                                    <a key={newsItem.uuid} href={newsItem.url} target={newsItem.target}>
                                        <time>{newsItem.dateTime}</time>
                                        <span className={newsItem.isRead ? 'is-read' : undefined}>{newsItem.title}</span>
                                    </a>
                                );
                            })
                        }
                    </nav>
                </Card>
            </>
        );
    }
}

const GlobalStyle = createGlobalStyle`
.fd-card-read-history {
    .h-read-history,
    > a,
    nav > a {
        padding: 12px 10px;
        @media only screen and (min-width: 641px) {
            padding: 12px;
        }
        @media only screen and (min-width: 861px) {
            padding: 12px 15px;
        }
    }

    > a,
    nav > a {
        display: block;
        font-family: 'ProximaNovaRegular', sans-serif;
        border-top: 1px solid rgba(0,0,0,0.1);
        line-height: 1.2em;
        color: #191919;
        .is-read {
            opacity: 0.6;
        }
        &:hover {
            color: #49a4a2;
            .icon-chevron-right {
                color: #49a4a2;
            }
        }
        text-decoration: none;
        font-size: 16px;
        @media only screen and (min-width: 641px) {
            font-size: 17px;
        }
    }

    time {
        color: #677381;
        display: block;
        font-size: 13px;
        @media only screen and (min-width: 641px) {
            font-size: 14px;
        }
    }
}
`;

export const ReadHistoryStyle = createGlobalStyle`
    ${(CardStyle as any).globalStyle.rules}
    ${getAllTextStyles(['card-h']).globalStyle.rules}
    ${(GlobalStyle as any).globalStyle.rules}
`;
