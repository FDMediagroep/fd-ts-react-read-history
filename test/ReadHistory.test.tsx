import TestRenderer from 'react-test-renderer';
import ReadHistory from '../src/ReadHistory';
import React from 'react';

describe('ReadHistory', () => {
    test('renders correctly', () => {
        let readHistory = TestRenderer.create(<ReadHistory news={[{uuid: '1', dateTime: '15:29', title: 'title van nieuwsbericht', url: 'https://fd.nl'}]}/>);
        expect(readHistory.toJSON()).toMatchSnapshot();
        readHistory = TestRenderer.create(<ReadHistory className="class-name" title="Recent gelezen blokje" news={[{uuid: '2', dateTime: '15:29', title: 'title van nieuwsbericht', url: 'https://fd.nl', target: "_blank"}]}/>);
        expect(readHistory.toJSON()).toMatchSnapshot();
    });
});
