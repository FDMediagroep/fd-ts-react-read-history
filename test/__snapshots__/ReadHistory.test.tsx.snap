// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadHistory renders correctly 1`] = `
<div
  className="fd-card fd-card-read-history"
>
  <h3
    className="fd-typography card-h h-read-history"
    textStyle="card-h"
  >
    Recent gelezen
  </h3>
  <nav>
    <a
      href="https://fd.nl"
    >
      <time>
        15:29
      </time>
      <span>
        title van nieuwsbericht
      </span>
    </a>
  </nav>
</div>
`;

exports[`ReadHistory renders correctly 2`] = `
<div
  className="fd-card fd-card-read-history class-name"
>
  <h3
    className="fd-typography card-h h-read-history"
    textStyle="card-h"
  >
    Recent gelezen blokje
  </h3>
  <nav>
    <a
      href="https://fd.nl"
      target="_blank"
    >
      <time>
        15:29
      </time>
      <span>
        title van nieuwsbericht
      </span>
    </a>
  </nav>
</div>
`;
